public/app/plugins/datasource/loki/components/useLokiLabels.ts

```diff
@@ -0,0 +1,75 @@
+import { useState, useEffect } from 'react';
+import LokiLanguageProvider from 'app/plugins/datasource/loki/language_provider';
+import { TypeaheadInput } from 'app/features/explore/QueryField';
+
+/**
+ *
+ * @param languageProvider
+ * @param languageProviderInitialised
+ * @param activeOption rc-cascader provided option used to fetch option's values that hasn't been loaded yet
+ *
+ * @description Fetches missing labels and enables labels refresh
+ */
+export const useLokiLabels = (
+  languageProvider: LokiLanguageProvider,
+  languageProviderInitialised: boolean,
+  activeOption: TypeaheadInput[]
```

Comment:
Shouldnt this be of type `CascaderOption[]`?

Reply:
Yes, absolutely. Need to investigate why TS did not complain about `onLoadOptions={setActiveOption}`, since `onLoadOptions` signature is :
```ts 
onLoadOptions: (selectedOptions: CascaderOption[]) => void;
```

Reply:
fixed in 5993adcf3c28dc87eb55d43ac9bf38cb869acbc7